src/client/services/ltik.js

```diff
@@ -1,22 +1,12 @@
-const getLtikPromise = new Promise((resolve, reject) => {
+const getLtik = () => {
   const searchParams = new URLSearchParams(window.location.search);
-  let potentialLtik = searchParams.get('ltik');
-  if (!potentialLtik) {
-    potentialLtik = sessionStorage.getItem('ltik');
-    if (!potentialLtik) reject(new Error('Missing lti key.'));
+  let ltik = searchParams.get('ltik');
+  if (!ltik) {
+    ltik = sessionStorage.getItem('ltik');
+    if (!ltik) throw new Error('Missing lti key.');

```

Comment:
Should setError be called here?

Reply:
The frontend would call setError when an error is thrown here

Reply:
It seems like error boundaries are not supported with React hooks, as they require getDerivedStateFromError and componentDidCatch: https://reactjs.org/docs/hooks-faq.html#do-hooks-cover-all-use-cases-for-classes

Reply:
cool otherwise looks good to me.